internal/pkg/cli/dependencies/repository.go

```diff
@@ -68,12 +70,72 @@ func (v *container) repositoryFs(definition model.TemplateRepository, template m
 		}
 		return aferofs.NewLocalFs(v.Logger(), path, definition.WorkingDir)
 	case model.RepositoryTypeGit:
-		return git.CheckoutTemplateRepositoryPartial(template, v.Logger())
+		return gitRepositoryFs(definition, template, v.Logger())
 	default:
 		panic(fmt.Errorf(`unexpected repository type "%s"`, definition.Type))
 	}
 }
 
+// gitRepositoryFs returns template repository FS, which has been loaded from a git repository.
+// Sparse checkout is used to load only the needed files.
+func gitRepositoryFs(definition model.TemplateRepository, tmplRef model.TemplateRef, logger log.Logger) (filesystem.Fs, error) {
+	// Checkout Git repository in sparse mode
+	gitRepository, err := git.Checkout(definition.Url, definition.Ref, true, logger)
```

Comment:
`git` pkg mi robil problemy, vznikali mi cyklicke zavislosti.
Hlavny problem bol v tom, ze `git` pkg nerobil iba Git tasky, ale pracoval aj priamo s `templates`,
to som vyriesil, a kod, ktory riesil templates je TU.

Je to v `cli/dependencies`, lebo sa to tyka iba CLI.
Jedna sa o `sparse checkout` repozitara, nacitavame iba tie subory, ktore potrebujeme.
Teda iba ohladom jednej sablony, ktoru ideme pouzit.